### PR TITLE
Modified user and question services on how they deal with failures when connecting to db

### PR DESCRIPTION
**Pull request by venuslimm**
- For user service: Added attempts to connect to MySQL db before stopping the docker container if MySQL db is not running
- For question service: Added code to stop question service container if cannot connect to Mongo atlas

### CODE REVIEW

**Review by lhy-hoyin — CHANGES_REQUESTED** [review]
Looks like there's some issues with `checkMySqlConnection`

File: User/src/UserServer.js [hunk 1]
```diff
@@ -16,6 +20,39 @@ const app = express();
 app.use(cors());
 app.use(bodyParser.json());
 app.use('/user', userRoutes);
-app.listen(env.USER_PORT, () => {
-  logger.log(`Running on port: ${env.USER_PORT}`);
-});
+
+const checkMySqlConnection = async () => {
```
[the remark this answers is not in the document]
Comment (venuslimm) [reply User/src/UserServer.js:24]: I am awaiting on lines 35 and 40 and would need it to be enclosed in an async function. I will be moving `app.listen` to be executing after `checkMySqlConnection` function

File: User/src/UserServer.js [hunk 2]
```diff
@@ -16,6 +20,39 @@ const app = express();
...
+  if (isMySqlRunning) {
+    app.listen(env.USER_PORT, () => {
+      logger.log(`Running on port: ${env.USER_PORT}`);
+    });
+  } else {
+    logger.error(
+      `${env.USER_STORAGE_CONTAINER_NAME} is not running on ${env.mysqlCreds.port}`
+    );
+    process.exit(1);
+  }
```
[the remark this answers is not in the document]
Comment (venuslimm) [reply User/src/UserServer.js:53]: Thanks for point that out, I will be replacing it with a more generic error message like unable to connect to MySQL

File: User/src/UserServer.js [hunk 3]
```diff
@@ -16,6 +20,39 @@ const app = express();
...
+    }
+  }
+
+  if (isMySqlRunning) {
+    app.listen(env.USER_PORT, () => {
+      logger.log(`Running on port: ${env.USER_PORT}`);
+    });
+  } else {
+    logger.error(
+      `${env.USER_STORAGE_CONTAINER_NAME} is not running on ${env.mysqlCreds.port}`
```
[the remark this answers is not in the document]
Comment (venuslimm) [reply User/src/UserServer.js:50]: Sorry, I think I might have accidentally deleted it when I was looking through my modifications before committing and pushing

**Review by lhy-hoyin — APPROVED** [review]
LGTM

**Pull request merged**